mscore/editstyle.cpp

```diff
@@ -665,6 +674,40 @@ EditStyle::EditStyle(Score* s, QWidget* parent)
       cs->startCmd();
       }
 
+//---------------------------------------------------------
+//   showEvent
+//---------------------------------------------------------
+
+void EditStyle::showEvent(QShowEvent* ev)
+      {
+      if (!hasShown && (isTooWide || isTooHigh)) {
+            // The default size is bigger than the screen, set an initial size
+            // that fits the screen.
+            // MuseScore might have restored a smaller size than the default,
+            // which is the only size that the constructor sees. So the size
+            // must be checked here too in order to allow MuseScore to restore
+            // the user's last size.
+            QRect scr = QGuiApplication::primaryScreen()->geometry();
+            QRect dlg = this->frameGeometry();
+            if (dlg.width() > scr.width() || dlg.height() > scr.height()) {
```

Comment:
Just to be sure, does something work wrong if we just disable the code inside this `if()` block? It looks like it forces window to fit to the screen size but it anyway gets done for me even if I disable this code. Moreover, even though this code tries to fit the window to a bit larger size than it should (see the comments on `availableSize`), it anyway gets corrected by Qt so the dialog still gets shown properly.

Reply:
On Debian 9 the window sizes correctly without this code?  Not on Windows or Ubuntu, not in my testing.  Is it only Debian 9 where you've seen this?  Try setting your screen size to 800x600 on Windows and see how it goes.

Reply:
Well, I must not have tested this since early in the process, because on Windows and Ubuntu, you are correct that this initial sizing code is not necessary.  Adding the QScrollArea seems to be enough to size the window to the screen.
This is an implicit, undocumented behavior on the part of Qt. That makes me nervous about removing the code.  Here's what I'm going to do:
I am commenting out this code in the next push.  I have another PR that should follow this one immediately.  If by the time that PR is ready to be merged, there are no problems reported, and we have tested at least on MacOS, then I will delete this commented code block in that next PR.  This way the code remains easily accessible in case the problem arises in that time frame?
Is that OK with you?  I'd prefer to leave the code commented that to stash it somewhere personal.  The next PR can be ready the very next day, and can finalize this.

Reply:
OK, I just tested one more thing.  Without the QScrollArea, on Ubuntu with only one 800x600 monitor, the window sizes correctly to the screen, but it truncates the contents.  I had originally tested on Ubuntu with both my monitors on, so the dialog sized wider.  But users are reporting that the dialog is not sizing to the screen, which is what I experienced initially, with 2 monitors.

To summarize: On Ubunty, this sizing to fit the screen has nothing to do with QScrollArea.  Therefore the original issue, as described by the users, is not reproducible on Ubuntu.  Now I must test this w/o the QScrollArea (again?) in Windows...

Reply:
I just tested on Windows and it's the same.  So it must be the minimum size that allows this to happen, and it's a standard Qt behavior?  I'll remove the initial sizing code completely and if no one else reports a problem we're fine.  Otherwise I'll stash it somewhere for later reinstatement.  I guess I got on the wrong track initially because of my initial testing results, and because I was not focused on minimum size, as you were.